forum/qa-plugin/tips-widget/tips-widget-widget.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+class tips_widget_widget
+{
+	public $random;
+	private $tips_list;
+	
+	function __construct()
+	{
+		if (!qa_opt('tips-enable'))
+			return;
+		
+		$this->tips_list = explode('!NEW!', qa_opt('tips-widget-content'));
+		$prev_random = isset($_COOKIE['prev_random']) ? $_COOKIE['prev_random'] : -1;
+		
+		do
+		{
+			$this->random = rand(0, count($this->tips_list) - 1);
+			
+		} while ($this->random == $prev_random);
+	}
+	
+	function allow_template($template)
+	{
+		if (!qa_opt('tips-enable'))
+			return false;
+		
+		return true;
+	}
+	
+	function allow_region($region)
+	{
+		if ($region == 'side')
+			return true;
+		
+		return false;
+	}
+	
+	function output_widget($region, $place, $themeobject, $template, $request, $qa_content)
+	{
+		if (!qa_opt('tips-enable'))
+			return;
+		
+		$themeobject->output('<div style="margin-bottom: 10px;"><b>Porady nie od parady</b></div>');
```

Comment:
po co wywoływanie 3x tej samej metody? nie można skonkatenować jakoś tego całego stringu skoro i tak jeden po drugim występuje?

Reply:
Zgadzam się, ale nie sądzę, aby powodowało to znaczne problemy z wydajnością. Po drugie taki zapis jest dość czytelny.